coast/index.py

```diff
@@ -0,0 +1,68 @@
+"""Index class."""
+from dask import array
+from dask.distributed import Client
+from .config import config_parser
+from .logging_util import get_slug, debug, info, warn, warning
+from typing import Union
+from pathlib import Path
+from ast import literal_eval
+
+
+def setup_dask_client(
+        workers: int = 2,
+        threads: int = 2,
+        memory_limit_per_worker: str = '2GB'
+):
+    Client(n_workers=workers, threads_per_worker=threads, memory_limit=memory_limit_per_worker)
+
+
+class Indexed:
+    def __init__(self, config: Union[Path, str] = None):
+        """ Configuration init.
+
+            Args:
+                config (Union[Path, str]): path to json config file.
+        """
+        debug(f"Creating a new {get_slug(self)}")
+
+        self.dataset = None
+        self.chunks = None
+        self.var_mapping = None
+        self.dim_mapping = None
+        self.coord_vars = None
+        self.keep_all_vars = False
+
+        if config:
+            self.json_config = config_parser.ConfigParser(config)
+            if self.json_config.config.chunks:
+                self.chunks = literal_eval(self.json_config.config.chunks[0])
```

Comment:
chunks will be a dictionary which can be passed directly into xarray/dask, so this indexing isn't required. You can just set self.chunks = self.json_config.config.chunks

(the indexing here will look for the key '0'  which may not always exist in a users chunks config)

Reply:
`print(type(self.json_config.config.chunks))`
returns
`<class 'list'>`

Then code errors with

```
File "/local/bodcweb/anaconda3/envs/coast/lib/python3.8/site-packages/xarray/backends/api.py", line 310, in _dataset_from_backend_dataset
    if not isinstance(chunks, (int, dict)) and chunks not in {None, "auto"}:
TypeError: unhashable type: 'list'
```

Reply:
> `print(type(self.json_config.config.chunks))`
> returns
> `<class 'list'>`
> 
> Then code errors with
> 
> ```
> File "/local/bodcweb/anaconda3/envs/coast/lib/python3.8/site-packages/xarray/backends/api.py", line 310, in _dataset_from_backend_dataset
>     if not isinstance(chunks, (int, dict)) and chunks not in {None, "auto"}:
> TypeError: unhashable type: 'list'
> ```


@qtl-bodc Yes it's because chunks is still a list in your config file json. It needs to be changed to just be a dict . There should be another comment specifying that change. :) 

i.e. it should now be:
```json
"chunks": {
        "time":1000
}
```

as oppose to :
```json
"chunks": [
		"{'time':1000}"
	]
```

Reply:
yes , got that now.